assets/src/domain/eventEditor/ui/tickets/ticketPriceCalculator/hooks/useMutatePrices.ts

```diff
@@ -0,0 +1,66 @@
+import { useCallback } from 'react';
+
+import { EntityId } from '@dataServices/types';
+import { cloneAndNormalizePrice } from '@sharedEntities/prices/predicates/updatePredicates';
+import { usePriceMutator } from '@edtrServices/apollo/mutations';
+import { TpcPriceModifier } from '../types';
+
+type Callback = (prices: Array<TpcPriceModifier>, deletedPrices: Array<EntityId>) => Promise<Array<EntityId>>;
+
+const useMutatePrices = (): Callback => {
+	const { createEntity: createPrice, deleteEntity: deletePrice, updateEntity: updatePrice } = usePriceMutator();
+
+	// Async to make sure that prices are handled before updating the ticket.
+	return useCallback(
+		async (prices, deletedPriceIds = []) => {
+			const relatedPriceIds: EntityId[] = [];
+
+			// make sure to complete all price mutatons before updating the ticket
+			await Promise.all(
+				// convert the price mutatons into promises
+				prices.map(({ isNew, isModified, ...price }) => {
+					// if it's not new or modified, no need to do anything
+					// but base price needs to be updated anyway which may been modified by revCalc
+					if (!(isNew || isModified) && !price.isBasePrice) {
+						// retain the existing relation
+						relatedPriceIds.push(price.id);
+						return Promise.resolve(price);
```

Comment:
in order to resolve a promise, a promise instance needs to be created first, which promise is resolved here?

Reply:
```
let promise = new Promise(function(resolve, reject) {
  // executor (the producing code, "singer")
});
```

Reply:
If the value passed to `Promise.resolve` is not thenable, then it's resolved with that value.

Reply:
hmm, interesting

Reply:
i didn't know that either but missed that in my review

Reply:
It's been there since beginning, I just extracted that code for re-usability.